krux_troposphere/troposphere.py

```diff
@@ -0,0 +1,169 @@
+    """
+    if not args:
+        parser = get_parser()
+        add_troposphere_cli_arguments(parser)
+        args = parser.parse_args()
+
+    if not logger:
+        logger = get_logger(name=NAME)
+
+    if not stats:
+        stats = get_stats(prefix=NAME)
+
+    boto = Boto3(
+        log_level=args.boto_log_level,
+        access_key=args.boto_access_key,
+        secret_key=args.boto_secret_key,
+        region=args.boto_region,
+        logger=logger,
+        stats=stats,
+    )
+    return Troposphere(
+        boto=boto,
+        logger=logger,
+        stats=stats,
+    )
+
+
+def add_troposphere_cli_arguments(parser, include_boto_arguments=True):
+    """
+    Utility function for adding Troposphere specific CLI arguments.
+    """
+    if include_boto_arguments:
+        # GOTCHA: Since many modules use krux_boto, the krux_boto's CLI arguments can be included twice,
+        # causing an error. This creates a way to circumvent that.
+
+        # Add all the boto arguments
+        add_boto_cli_arguments(parser)
+
+    # Add those specific to the application
+    group = get_group(parser, NAME)
+
+
+class Troposphere(object):
+    """
+    A manager to handle all Troposphere / Cloud Formation related functions.
+    Each instance is locked to a connection to a designated region (self.boto.cli_region).
+    """
+
+    STACK_NOT_EXIST_ERROR_MSG = 'Stack with id {stack_name} does not exist'
+    NO_UPDATE_ERROR_MSG = 'No updates are to be performed.'
+
+    def __init__(
+        self,
+        boto,
+        logger=None,
+        stats=None,
+    ):
+        """
+        :param boto: :py:class:`krux_boto.boto.Boto3` Boto3 object used to connect to Cloud Formation
+        :param logger: :py:class:`logging.Logger` Logger, recommended to be obtained using krux.cli.Application
+        :param stats: :py:class:`kruxstatsd.StatsClient` Stats, recommended to be obtained using krux.cli.Application
+        """
+        # Private variables, not to be used outside this module
+        self._name = NAME
+        self._logger = logger or get_logger(self._name)
+        self._stats = stats or get_stats(prefix=self._name)
+
+        if not isinstance(boto, Boto3):
+            raise NotImplementedError('Currently krux_troposphere.troposphere.Troposphere only supports krux_boto.boto.Boto3')
+
+        self._cf = boto.client('cloudformation')
+        self.template = troposphere.Template()
+
+    def _is_stack_exists(self, stack_name):
+        """
+        Check if the given Cloud Formation stack exists
+
+        GOTCHA: There is no simple way to check the existence of a stack.
+        The template for the stack is fetched and if an expected exception occur (Unable to find stack),
+        then the stack is deemed not existing.
+
+        :param stack_name: :py:class:`str` Name of the stack to check
+        """
+        try:
+            # See if we can get a template for this
+            self._cf.get_template(StackName=stack_name)
```

Comment:
`list_stacks()` will probably return a lot more data than I want. Also, I will need to deal with pagination. Thus, I moved away from it. I'm happy to change my way.